### PR TITLE
refactor: add simillar labels to the each components

### PR DESCRIPTION
**Pull request by iamniting**
Add labels to the topolvm-controller, topolvm-node, vg-manager
for example:
app.lvm.openshift.io: topolvm-controller
app.lvm.openshift.io: topolvm-node
app.lvm.openshift.io: vg-manager

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: controllers/topolvm_controller.go
Comment (iamniting) [inline controllers/topolvm_controller.go:140]: Are we using this existing label anywhere? If not I would like to remove it.

File: controllers/topolvm_node.go
Comment (iamniting) [inline controllers/topolvm_node.go:179]: Are we using this existing label anywhere? If not I would like to remove it.

File: controllers/vgmanager_daemonset.go
Comment (iamniting) [inline controllers/vgmanager_daemonset.go:174]: Are we using this existing label anywhere? If not I would like to remove it.

File: controllers/topolvm_controller.go
Comment (leelavg) [reply controllers/topolvm_controller.go:140]: can be removed

File: controllers/topolvm_node.go
Comment (leelavg) [reply controllers/topolvm_node.go:179]: can be removed

File: controllers/vgmanager_daemonset.go
Comment (leelavg) [reply controllers/vgmanager_daemonset.go:174]: can be removed

**Review by leelavg — COMMENTED** [review]
- pardon, forgot to mention earlier, we might also need a label on manager itself.
- in addition to this label if we can have`app.kubernetes.io/part-of: lvmo` as well then we can get all possible resources under lvmo using a single label.

**Comment by iamniting** [comment]
/hold

Require discussion before merging the PR, As this will break the upgrades.

```
The Deployment "nginx-deployment" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"nginx", "nigoyal":"nigoyal"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable```

**Comment by iamniting** [comment]
> * pardon, forgot to mention earlier, we might also need a label on manager itself.

That is a topic of discussion and should be taken care of separately as that is a CSV change and will break upgrades very badly.

> * in addition to this label if we can have`app.kubernetes.io/part-of: lvmo` as well then we can get all possible resources under lvmo using a single label.

We can even list all of them without adding the above labels using only key I added in the PR for example
```
$ oc get pod -l app
NAME                                READY   STATUS    RESTARTS   AGE
nginx-deployment-74dc8cc494-b4fpn   1/1     Running   0          21m
topolvm-node-lgps4                  4/4     Running   0          2d19h
vg-manager-7k5lf                    1/1     Running   0          2d19h
```

**Comment by iamniting** [comment]
```
┌─[nigoyalॐ nigoyal ~/code/go/src/github.com/red-hat-storage/lvm-operator  labels]
└──╼$ oc get pods -l app.lvm.topolvm.io
NAME                                  READY   STATUS    RESTARTS   AGE
topolvm-controller-7546d896f6-rqkgh   4/4     Running   0          3m41s
topolvm-node-ljt7d                    4/4     Running   0          3m41s
vg-manager-2mcj2                      1/1     Running   0          3m41s

┌─[nigoyalॐ nigoyal ~/code/go/src/github.com/red-hat-storage/lvm-operator  labels]
└──╼$ oc get pods -l app.lvm.topolvm.io=topolvm-controller
NAME                                  READY   STATUS    RESTARTS   AGE
topolvm-controller-7546d896f6-rqkgh   4/4     Running   0          3m48s

┌─[nigoyalॐ nigoyal ~/code/go/src/github.com/red-hat-storage/lvm-operator  labels]
└──╼$ oc get pods -l app.lvm.topolvm.io=topolvm-node
NAME                 READY   STATUS    RESTARTS   AGE
topolvm-node-ljt7d   4/4     Running   0          4m10s

┌─[nigoyalॐ nigoyal ~/code/go/src/github.com/red-hat-storage/lvm-operator  labels]
└──╼$ oc get pods -l app.lvm.topolvm.io=vg-manager
NAME               READY   STATUS    RESTARTS   AGE
vg-manager-2mcj2   1/1     Running   0          4m14s
```

File: controllers/defaults.go
Comment (nbalacha) [inline controllers/defaults.go:108]:
```suggestion
	DefaultLabelKey           = "app.lvm.openshift.io"
```

Comment (iamniting) [reply controllers/defaults.go]: done

**Comment by iamniting** [comment]
> Please add a label to the main controller-manager as well.

I would like to take it as separate PR as the changes are in the kustomize and related files and not in the controller.

**Review by nbalacha — APPROVED** [review]

**Pull request merged**
